packages/flutter_test/lib/src/binding.dart

```diff
@@ -373,11 +373,33 @@ abstract class TestWidgetsFlutterBinding extends BindingBase
   /// [AutomatedTestWidgetsFlutterBinding] implementation to increase the
   /// current timeout, if any. See [AutomatedTestWidgetsFlutterBinding.addTime]
   /// for details.
+  ///
+  /// If your test code depends on the result of a task that did not complete,
+  /// the most common symptom would be a hanging test as FakeAsync environment
+  /// is unable to resolve a future that was created in `runAsync`. If you
+  /// observe such behavior or flakiness, you have a number of options:
+  ///
+  /// - Consider restructuring your code so you do not need `runAsync`. This is
+  ///   the optimal solution as widget tests are designed to run in Fake async
+  ///   environment.
+  ///
+  /// - Expose a [Future] in your application code that signals the readiness of
+  ///   your widget tree, then await that future inside `callback`.
+  ///
+  /// - If either option is not feasible, await [resolveFutures] inside
+  ///   `callback`.
   Future<T> runAsync<T>(
     Future<T> callback(), {
     Duration additionalTime = const Duration(milliseconds: 1000),
   });
 
+  /// Waits for a stable state within a single `runAsync` execution.
+  ///
+  /// Note that using this method is *not* recommended. This method should only
```

Comment:
please see the style guide for writing style for documentation (https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo) -- e.g. "Note that" is against our style.